source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -5906,13 +5906,17 @@ export abstract class StylableCardElementContainer extends CardElementContainer
     }
 
     protected getHasBackground(): boolean {
+        return this.getHasPadding(true);
```

Comment:
This can't be. We can't determine that something has a background because it has padding when we determine that it has padding because it has a background

Reply:
To be clear, the logic has so far been that something (typically a container, but the JS SDK could technically support that on any element) has padding because it either has a background (via a container style or a background image). The base implementation in CardElement cannot return `true`. And I don't see any reason why the base implementation should be changed from its current implementation.

Reply:
Plus the implementation of `StylableCardElementContainer.getDefaultPadding` is wonky:

```typescript
protected getDefaultPadding(): PaddingDefinition {
        let hasPadding = GlobalSettings.addPaddingToContainerWithBackgroundImage ? 
            this.getHasBackground() : this.getHasPadding(false);
        ...
```

It should be
```typescript
let hasPadding = this.getHasPadding(GlobalSettings.addPaddingToContainerWithBackgroundImage);
```

Reply:
@dclaux; before the change getHasBackground() of `StyleableCardElementContainer` doesn't consider background image of current element since it start with current element's parent. we could change that to current, but I don't know about the side effects if I do. the reason that it has been working is because this.getHasBackground calls the Container's method which checks if the container's background is valid, only if fails it calls getHasBackground of `StylableCardElementContainer`.